client/widgets/markets/TradePanels.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * TradePanels.h, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#pragma once
+
+#include "../Images.h"
+
+#include "../../../lib/networkPacks/TradeItem.h"
+
+class CTextBox;
+class SelectableSlot;
+
+enum class EType
+{
+	RESOURCE, PLAYER, ARTIFACT_TYPE, CREATURE, CREATURE_PLACEHOLDER, ARTIFACT_PLACEHOLDER, ARTIFACT_INSTANCE
+};
+
+class CTradeableItem : public CIntObject, public std::enable_shared_from_this<CTradeableItem>
+{
+public:
+	std::shared_ptr<CAnimImage> image;
+	AnimationPath getFilename();
+	int getIndex();
+	using ClickPressedFunctor = std::function<void(const std::shared_ptr<CTradeableItem>&)>;
+
+	const CArtifactInstance * hlp; //holds ptr to artifact instance id type artifact
+	EType type;
+	int id;
+	const int serial;
+	const bool left;
+	std::string subtitle;
+	ClickPressedFunctor clickPressedCallback;
+
+	void setType(EType newType);
+	void setID(int newID);
+
+	const CArtifactInstance * getArtInstance() const;
+	void setArtInstance(const CArtifactInstance * art);
+
+	std::unique_ptr<SelectableSlot> selection;
+	bool downSelection;
+
+	void showAllAt(const Point & dstPos, const std::string & customSub, Canvas & to);
+
+	void showPopupWindow(const Point & cursorPosition) override;
+	void hover(bool on) override;
+	void showAll(Canvas & to) override;
+	void clickPressed(const Point & cursorPosition) override;
+	std::string getName(int number = -1) const;
+	CTradeableItem(Point pos, EType Type, int ID, bool Left, int Serial);
+};
+
+struct STradePanel : public CIntObject
```

Comment:
- should probably be a class, it is strange IMO for struct to inherit from struct (or vice-versa)
- we don't use 'S' or 'C' prefixes for new structs / classes

(same for other entities in this file)
```suggestion
class TradePanel : public CIntObject
```

Reply:
> * we don't use 'S' or 'C' prefixes for new structs / classes

Noted